<launch>
  <!-- ===== Args ===== -->
  <arg name="map_frame" default="map"/>
  <arg name="base_frame" default="base_link"/>
  <arg name="waypoints_file" default="$(find limo_mission_manager)/params/waypoints.yaml"/>
  <arg name="loop" default="true"/>                 <!-- true면 순환 -->
  <arg name="start_index" default="0"/>
  <arg name="goal_timeout" default="120.0"/>        <!-- 초 -->
  <arg name="retry_limit" default="2"/>             <!-- 실패 시 재시도 횟수 -->
  <arg name="clear_costmap_on_fail" default="true"/>
  <arg name="backup_on_fail" default="true"/>       <!-- 실패 시 살짝 후진 -->
  <arg name="backup_vel" default="-0.08"/>          <!-- m/s -->
  <arg name="backup_time" default="1.0"/>           <!-- s -->
  <arg name="cmd_vel_topic" default="/cmd_vel"/>

  <!-- 미션 매니저 -->
  <node pkg="limo_mission_manager" type="mission_manager.py" name="mission_manager" output="screen">
    <param name="map_frame" value="$(arg map_frame)"/>
    <param name="base_frame" value="$(arg base_frame)"/>
    <param name="waypoints_file" value="$(arg waypoints_file)"/>
    <param name="loop" value="$(arg loop)"/>
    <param name="start_index" value="$(arg start_index)"/>
    <param name="goal_timeout" value="$(arg goal_timeout)"/>
    <param name="retry_limit" value="$(arg retry_limit)"/>
    <param name="clear_costmap_on_fail" value="$(arg clear_costmap_on_fail)"/>
    <param name="backup_on_fail" value="$(arg backup_on_fail)"/>
    <param name="backup_vel" value="$(arg backup_vel)"/>
    <param name="backup_time" value="$(arg backup_time)"/>
    <param name="cmd_vel_topic" value="$(arg cmd_vel_topic)"/>
  </node>

  <!-- 모니터(진행/거리/상태 진단) -->
  <node pkg="limo_mission_manager" type="goal_monitor.py" name="goal_monitor" output="screen">
    <param name="map_frame" value="$(arg map_frame)"/>
    <param name="base_frame" value="$(arg base_frame)"/>
  </node>
</launch>
